### DISCUSSION TITLE
Exporting a custom DG (not DAG) node.

### DISCUSSION THREAD

**Discussion by jjzanin-SMS** [Autodesk/maya-usd#3140] [Q&A]
Hello,

Is it possible to export a custom DG (not DAG) node inheriting from UsdMayaPrimWriter? It looks like UsdMaya_WriteJob::_BeginWriting only traverses the DAG, so nodes that aren't DAG nodes are skipped. The comment for UsdMayaPrimWriter says it writes prims for DG nodes too:

/// Base class for all built-in and user-defined prim writers. Translates Maya
/// node data into USD prim(s).
///
/// **Note that this class can be used to write USD prims for both DG and DAG
/// Maya nodes**. For DAG nodes, an MDagPath is required to uniquely identify
/// instances in the DAG, so the writer should be created using an
/// MFnDagNode (or one of its derived classes) that was constructed using an
/// MDagPath, *not* an MObject.

Thanks!
Jeremiah

**Comment by gelshoffNV** [comment]
It will traverse the DAG, but translate any DG connections, but if you want to translate something not connected to the DAG, you may either need to connect it or handle the translation in a chaser...

**Reply by jjzanin-SMS** [reply]
That sounds great, can you point me to where it translates DG nodes through connections? I just see the UsdMayaPrimWriter::Write function being called in the DAG iteration in UsdMaya_WriteJob::_BeginWriting here: https://github.com/Autodesk/maya-usd/blob/8ec91f942bf6d6ff82c636c17406b600c83b9bbb/lib/mayaUsd/fileio/jobs/writeJob.cpp#LL397C35-L397C36
